aquadoggo/src/network/transport.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use futures::future::Either;
+use libp2p::core::muxing::StreamMuxerBox;
+use libp2p::core::transport::upgrade::Version;
+use libp2p::core::transport::{Boxed, OrTransport};
+use libp2p::identity::Keypair;
+use libp2p::noise::NoiseAuthenticated;
+use libp2p::yamux::YamuxConfig;
+use libp2p::{quic, relay, PeerId, Transport};
+
+// Build the transport stack to be used by the network swarm
+pub async fn build_transport(
+    key_pair: &Keypair,
+    relay_client_enabled: bool,
+) -> (
+    Boxed<(PeerId, StreamMuxerBox)>,
+    Option<relay::client::Behaviour>,
+) {
+    // Create QUIC transport (provides transport, security and multiplexing in a single protocol)
+    let quic_config = quic::Config::new(key_pair);
+    let quic_transport = quic::tokio::Transport::new(quic_config);
+
+    let (transport, relay_client) = if relay_client_enabled {
+        // Generate a relay transport and client behaviour if relay client mode is enabled
+        let (relay_transport, relay_client) = relay::client::new(key_pair.public().to_peer_id());
+        let relay_client = Some(relay_client);
+
+        // Add encryption and multiplexing to the relay transport
+        let relay_transport = relay_transport
+            .upgrade(Version::V1)
+            .authenticate(NoiseAuthenticated::xx(key_pair).unwrap())
+            .multiplex(YamuxConfig::default());
+
+        // The appropriate transport will be matched and utilised for connections (ie.
+        // relayed-connections will utilise the relay transport while all other incoming
```

Comment:
Does this mean relayed connections don't use QUIC (did we already discuss this, i feel like it was supported already...)?

Reply:
https://github.com/libp2p/rust-libp2p/discussions/3766

Reply:
As confirmed by Thomas, the relayed connections will use QUIC in our case. I'm going to quickly alter some of the variable naming to make this clearer in the code.